src/Fhir.Anonymizer.Core/Extensions/ElementNodeNavExtensions.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Hl7.Fhir.ElementModel;
+using Hl7.FhirPath;
+
+namespace Fhir.Anonymizer.Core.Extensions
+{
+    public static class ElementNodeNavExtensions
+    {
+        public static List<ElementNode> GetEntryResourceChildren(this ElementNode node)
+        {
+            return node?.Children(Constants.EntryNodeName)
+                    .Select(entry => entry?.Children(Constants.ResourceNodeName).FirstOrDefault())
+                    .Where(resource => resource != null)
+                    .Cast<ElementNode>()
+                    .ToList();
+        }
+
+        public static List<ElementNode> GetContainedChildren(this ElementNode node)
+        {
+            return node?.Children(Constants.ContainedNodeName).Cast<ElementNode>().ToList();
+        }
+
+        public static IEnumerable<ElementNode> ResourceDescendants(this ElementNode node)
```

Comment:
Should we add "WithoutSubResources" in this func name too (similar to func below)?

Reply:
Thanks, added.